core/src/main/java/org/apache/carbondata/core/metadata/SegmentFileStore.java

```diff
@@ -1125,19 +1125,21 @@ public static void deleteSegment(String tablePath, Segment segment,
     SegmentFileStore fileStore = new SegmentFileStore(tablePath, segment.getSegmentFileName());
     List<String> indexOrMergeFiles = fileStore.readIndexFiles(SegmentStatus.SUCCESS, true,
         FileFactory.getConfiguration());
-    Map<String, List<String>> indexFilesMap = fileStore.getIndexFilesMap();
-    for (Map.Entry<String, List<String>> entry : indexFilesMap.entrySet()) {
-      FileFactory.deleteFile(entry.getKey());
-      for (String file : entry.getValue()) {
-        String[] deltaFilePaths =
-            updateStatusManager.getDeleteDeltaFilePath(file, segment.getSegmentNo());
-        for (String deltaFilePath : deltaFilePaths) {
-          FileFactory.deleteFile(deltaFilePath);
+    if (!indexOrMergeFiles.isEmpty() && indexOrMergeFiles.get(0).startsWith(tablePath)) {
```

Comment:
what if the external segment is copied inside table folder, this will delete external segment also

Reply:
I have modified the check for deleting physical data and also added the testcases for this scenario.